--- texlive-20150521-source/libs/luajit/LuaJIT-2.0.3/src/Makefile.omv~	2016-05-06 12:46:38.878198696 +0200
+++ texlive-20150521-source/libs/luajit/LuaJIT-2.0.3/src/Makefile	2016-05-06 12:47:26.311045852 +0200
@@ -196,6 +196,7 @@ TARGET_DLLNAME= lua$(NODOTABIVER).dll
 TARGET_XSHLDFLAGS= -shared -fPIC -Wl,-soname,$(TARGET_SONAME)
 TARGET_DYNXLDOPTS=
 
+TARGET_LIBS= -lgcc_s
 TARGET_LFSFLAGS= -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 TARGET_XCFLAGS= $(TARGET_LFSFLAGS) -U_FORTIFY_SOURCE
 TARGET_XLDFLAGS=
--- texlive-20150521-source/libs/luajit/Makefile.am.omv~	2016-05-06 12:48:31.054834339 +0200
+++ texlive-20150521-source/libs/luajit/Makefile.am	2016-05-06 12:49:16.509684045 +0200
@@ -24,6 +24,7 @@ SUBDIRS = . native include
 lib_LTLIBRARIES = libtexluajit.la
 
 libtexluajit_la_LDFLAGS = -bindir @bindir@ -no-undefined -version-info $(LUAJIT_LT_VERSINFO)
+libtexluajit_la_LIBADD = -lgcc_s
 
 nodist_libtexluajit_la_SOURCES = \
 	$(ljlib_sources) \
