diff -up texlive-20180414-source/texk/web2c/synctexdir/synctex_parser.c.omv~ texlive-20180414-source/texk/web2c/synctexdir/synctex_parser.c
--- texlive-20180414-source/texk/web2c/synctexdir/synctex_parser.c.omv~	2018-05-20 16:37:55.311807102 +0200
+++ texlive-20180414-source/texk/web2c/synctexdir/synctex_parser.c	2018-05-20 16:38:26.318110840 +0200
@@ -83,6 +83,8 @@
  *  First level objects are sheets and forms, containing boxes, glues, kerns...
  *  The third tree allows to browse leaves according to tag and line.
  */
+#define _GNU_SOURCE 1 // for vasprintf
+
 #   if defined(SYNCTEX_USE_LOCAL_HEADER)
 #       include "synctex_parser_local.h"
 #   else
