--- texlive-20140525-source/texk/mendexk/fwrite.c.omv~	2014-06-17 10:51:45.828093027 +0200
+++ texlive-20140525-source/texk/mendexk/fwrite.c	2014-06-17 10:52:05.958093092 +0200
@@ -17,8 +17,8 @@ static int range_check(struct index ind,
 static void linecheck(char *lbuff, char *tmpbuff);
 static void crcheck(char *lbuff, FILE *fp);
 
-/*  OS X defines snprintf() as macro that chokes on using TAIL_LEN  */
-#ifdef __APPLE__
+/* Clang defines snprintf() as macro that chokes on using TAIL_LEN  */
+#ifdef __clang__
 #undef snprintf
 #endif
 
