diff -up texlive-20230311-source/utils/asymptote/settings.cc.1~ texlive-20230311-source/utils/asymptote/settings.cc
--- texlive-20230311-source/utils/asymptote/settings.cc.1~	2023-04-15 14:38:31.843005548 +0200
+++ texlive-20230311-source/utils/asymptote/settings.cc	2023-04-15 14:40:42.882095720 +0200
@@ -101,9 +101,9 @@ string defaultPSViewer="open";
 string defaultPDFViewer="open";
 string defaultHTMLViewer="open";
 #else
-string defaultPSViewer="evince";
-string defaultPDFViewer="evince";
-string defaultHTMLViewer="google-chrome";
+string defaultPSViewer="xdg-open";
+string defaultPDFViewer="xdg-open";
+string defaultHTMLViewer="xdg-open";
 #endif
 string defaultGhostscript="gs";
 string defaultGhostscriptLibrary="";
