diff -up texlive-20210324-source/texk/web2c/pdftexdir/pdftoepdf.cc.omv~ texlive-20210324-source/texk/web2c/pdftexdir/pdftoepdf.cc
--- texlive-20210324-source/texk/web2c/pdftexdir/pdftoepdf.cc.omv~	2022-03-28 21:04:19.076106289 +0200
+++ texlive-20210324-source/texk/web2c/pdftexdir/pdftoepdf.cc	2022-03-28 21:13:05.517671957 +0200
@@ -158,8 +158,7 @@ static PdfDocument *find_add_document(ch
     p->file_name = xstrdup(file_name);
     p->xref = xref = 0;
     p->occurences = 0;
-    GString *docName = new GString(p->file_name);
-    p->doc = new PDFDoc(docName);       // takes ownership of docName
+    p->doc = new PDFDoc(std::make_unique<GString>(p->file_name));
     if (!p->doc->isOk() || !p->doc->okToPrint()) {
         pdftex_fail("xpdf: reading PDF image failed");
     }
diff -up texlive-20210324-source/texk/web2c/pdftexdir/pdftosrc.cc.omv~ texlive-20210324-source/texk/web2c/pdftexdir/pdftosrc.cc
--- texlive-20210324-source/texk/web2c/pdftexdir/pdftosrc.cc.omv~	2022-03-28 21:13:26.781746942 +0200
+++ texlive-20210324-source/texk/web2c/pdftexdir/pdftosrc.cc	2022-03-28 21:14:27.065962761 +0200
@@ -64,7 +64,6 @@ int main(int argc, char *argv[])
 {
     char *p, buf[1024];
     PDFDoc *doc;
-    GString *fileName;
     Stream *s;
     Object srcStream, srcName, catalogDict;
     FILE *outfile;
@@ -78,9 +77,8 @@ int main(int argc, char *argv[])
                 "Usage: pdftosrc <PDF-file> [<stream-object-number>]\n");
         exit(1);
     }
-    fileName = new GString(argv[1]);
     globalParams = std::make_unique<GlobalParams>();
-    doc = new PDFDoc(fileName);
+    doc = new PDFDoc(std::make_unique<GString>(argv[1]));
     if (!doc->isOk()) {
         fprintf(stderr, "Invalid PDF file\n");
         exit(1);
@@ -118,7 +116,7 @@ int main(int argc, char *argv[])
             fprintf(stderr, "Not a Stream object\n");
             exit(1);
         }
-        sprintf(buf, "%s", fileName->c_str());
+        sprintf(buf, "%s", argv[1]);
         if ((p = strrchr(buf, '.')) == 0)
             p = strchr(buf, 0);
         if (objgen == 0)
@@ -128,7 +126,7 @@ int main(int argc, char *argv[])
         outname = buf;
     } else {                    // objnum < 0 means we are extracting the XRef table
         extract_xref_table = true;
-        sprintf(buf, "%s", fileName->c_str());
+        sprintf(buf, "%s", argv[1]);
         if ((p = strrchr(buf, '.')) == 0)
             p = strchr(buf, 0);
         sprintf(p, ".xref");
