--- texlive-20140525-source/texk/kpathsea/texmf.cnf.p3~	2014-06-16 23:48:57.314964931 +0200
+++ texlive-20140525-source/texk/kpathsea/texmf.cnf	2014-06-16 23:50:40.201965262 +0200
@@ -450,67 +450,7 @@ RUBYINPUTS   = .;$TEXMF/scripts/{$progna
 %% The mktex* scripts rely on KPSE_DOT. Do not set it in the environment.
 % KPSE_DOT = .
 
-% This definition isn't used from this .cnf file itself (that would be
-% paradoxical), but the compile-time default in paths.h is built from it.
-% The SELFAUTO* variables are set automatically from the location of
-% argv[0], in kpse_set_program_name.
-% 
-% This main texmf.cnf file is installed, for a release YYYY, in a
-% directory such as /usr/local/texlive/YYYY/texmf-dist/web2c/texmf.cnf.
-% Since this file is subject to future updates, the TeX Live installer
-% or human administrator may also create a file
-% /usr/local/texlive/YYYY/texmf.cnf; any settings in this latter file
-% will take precedence over the distributed one.
-% 
-% For security reasons, it is best not to include . in this path.
-%
-% The idea behind this lengthy definition: for each of
-% SELFAUTO{LOC,DIR,PARENT}, look in the directory,
-% then the subdirectories share/texmf-local, share/texmf-dist, share/texmf,
-% then the subdirectories ./texmf-local, texmf-dist, ./texmf.
-% At any given installation, most of these directories will not exist,
-% but they all turn out to be useful somewhere.
-% 
-% Special addition -- we want to include one more directory: the
-% great-grandparent's texmf-local, because that is how TL is installed
-% by default.  That is, given a binary
-% /usr/local/texlive/YYYY/bin/PLATFORM/kpsewhich, it should find
-% /usr/local/texlive/texmf-local/web2c/texmf.cnf.  I.e., not under YYYY.
-% 
-% As a result, we cannot use actual brace expansion in the definition,
-% since we don't want to scatter ../'s throughout the value.  Hence we
-% explicitly list every directory.  Arguably more understandable anyway.
-%
-TEXMFCNF = {\
-$SELFAUTOLOC,\
-$SELFAUTOLOC/share/texmf-local/web2c,\
-$SELFAUTOLOC/share/texmf-dist/web2c,\
-$SELFAUTOLOC/share/texmf/web2c,\
-$SELFAUTOLOC/texmf-local/web2c,\
-$SELFAUTOLOC/texmf-dist/web2c,\
-$SELFAUTOLOC/texmf/web2c,\
-\
-$SELFAUTODIR,\
-$SELFAUTODIR/share/texmf-local/web2c,\
-$SELFAUTODIR/share/texmf-dist/web2c,\
-$SELFAUTODIR/share/texmf/web2c,\
-$SELFAUTODIR/texmf-local/web2c,\
-$SELFAUTODIR/texmf-dist/web2c,\
-$SELFAUTODIR/texmf/web2c,\
-\
-$SELFAUTOGRANDPARENT/texmf-local/web2c,\
-$SELFAUTOPARENT,\
-\
-$SELFAUTOPARENT/share/texmf-local/web2c,\
-$SELFAUTOPARENT/share/texmf-dist/web2c,\
-$SELFAUTOPARENT/share/texmf/web2c,\
-$SELFAUTOPARENT/texmf-local/web2c,\
-$SELFAUTOPARENT/texmf-dist/web2c,\
-$SELFAUTOPARENT/texmf/web2c\
-}
-%
-% For reference, here is the old brace-using definition:
-%TEXMFCNF = {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT}{,{/share,}/texmf{-local,}/web2c}
+TEXMFCNF = @@TEXMFCNF@@
 
 % kpathsea 3.5.3 and later sets these at runtime. To avoid empty
 % expansions from binaries linked against an earlier version of the
