--- texlive-20120810-source/texk/xdvik/xdvi-sh.in.orig	2012-08-10 13:56:02.355764330 -0300
+++ texlive-20120810-source/texk/xdvik/xdvi-sh.in	2012-08-10 13:56:53.408763037 -0300
@@ -4,19 +4,6 @@
 # Copyright Thomas Esser, 1998. Permission to distribute under the terms
 # of the GNU general public license version 2 or later.
 
-# This script sets some environment variables to make sure that xdvi's
-# resource file in $XDVIINPUTS/xdvi is read by xdvi.bin.
-
-test -f /bin/sh5 && test -z "$RUNNING_SH5" \
-  && test x"`(uname -s) 2>&1`" = xULTRIX \
-  && { RUNNING_SH5=true; export RUNNING_SH5; exec /bin/sh5 $0 ${1+"$@"}; }
-RUNNING_SH5=
-
-test -f /bin/bsh && test -z "$RUNNING_BSH" \
-  && test x"`(uname -s) 2>&1`" = xAIX \
-  && { RUNNING_BSH=true; export RUNNING_BSH; exec /bin/bsh $0 ${1+"$@"}; }
-RUNNING_BSH=
-
 # -help and -version only used to work if they were the only options,
 # so the "-name xdvi" parameter wasn't used in that case. With current
 # xdvik, it should work in any case, but we keep this logic for compatibility
@@ -35,19 +22,4 @@ case "$#:$1" in
     NAMEOPT="-name $BASE_NAME";;
 esac
 
-xdviappfile=`kpsewhich -progname=xdvi --format='other text files' XDvi`
-if test -n "$xdviappfile"; then
-  xdviappdir=`dirname "$xdviappfile"`
-  xdviapppath="${xdviappdir}/%l_%t/%N:${xdviappdir}/%N"
-
-  # For R3, we have to set XAPPLRESDIR.
-  ### SU: removed, xdvik won't work with R3 anyway, and XAPPLRESDIR is
-  ### for user customizations in current X
-  ### XAPPLRESDIR="$xdviappdir"; export XAPPLRESDIR
-
-  # For R4 or later, we have to set XFILESEARCHPATH, since XAPPLRESDIR might
-  # be ignored (if XUSERFILESEARCHPATH is set)
-  XFILESEARCHPATH="$xdviapppath:${XFILESEARCHPATH-%D}"; export XFILESEARCHPATH
-fi
-
 exec @final_exec_name@ $NAMEOPT ${1+"$@"}
--- texlive-20120810-source/utils/asymptote/settings.cc.orig	2012-08-10 13:53:33.856768325 -0300
+++ texlive-20120810-source/utils/asymptote/settings.cc	2012-08-10 13:54:26.048766966 -0300
@@ -100,7 +100,7 @@ string defaultPSViewer="gv";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
 string defaultDisplay="display";
@@ -1351,9 +1351,9 @@ string lookup(const string& symbol)
 
 void initDir() {
   if(getSetting<string>("sysdir").empty()) {
-    string s=lookup("SELFAUTOPARENT");
+    string s=lookup("TEXMFMAIN");
     if(s.size() > 1) {
-      string texmf=s+dirsep+"texmf"+dirsep;
+      string texmf=s+dirsep;
       docdir=texmf+"doc"+dirsep+"asymptote";
       Setting("sysdir")=texmf+"asymptote";
       s=lookup("ASYMPTOTE_HOME");
