diff -up texlive-20190413-source/utils/asymptote/settings.cc.1~ texlive-20190413-source/utils/asymptote/settings.cc
--- texlive-20190413-source/utils/asymptote/settings.cc.1~	2019-04-21 17:37:33.503903464 +0200
+++ texlive-20190413-source/utils/asymptote/settings.cc	2019-04-21 17:38:27.775900831 +0200
@@ -95,7 +95,7 @@ string defaultPSViewer="gv";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
 string defaultGhostscriptLibrary="";
