--- texlive-20120810-source/texk/ptexenc/Makefile.am.orig	2012-08-10 13:35:17.762797903 -0300
+++ texlive-20120810-source/texk/ptexenc/Makefile.am	2012-08-10 13:35:40.627797324 -0300
@@ -18,6 +18,8 @@ libptexenc_la_CPPFLAGS = -DMAKE_PTENC_DL
 
 libptexenc_la_LDFLAGS = -no-undefined -version-info $(PTEXENC_LT_VERSINFO)
 
+libptexenc_la_LIBADD = ../kpathsea/libkpathsea.la
+
 # Note: jisx0208.h is a copy of ../../libs/gd/gd-2.0.35/jisx0208.h.
 #
 libptexenc_la_SOURCES = \
--- texlive-20120810-source/texk/ptexenc/Makefile.in.orig	2012-08-10 13:35:49.806796922 -0300
+++ texlive-20120810-source/texk/ptexenc/Makefile.in	2012-08-10 13:36:07.007796530 -0300
@@ -291,6 +291,7 @@ AM_CFLAGS = $(WARNING_CFLAGS)
 lib_LTLIBRARIES = libptexenc.la
 libptexenc_la_CPPFLAGS = -DMAKE_PTENC_DLL
 libptexenc_la_LDFLAGS = -no-undefined -version-info $(PTEXENC_LT_VERSINFO)
+libptexenc_la_LIBADD = ../kpathsea/libkpathsea.la
 
 # Note: jisx0208.h is a copy of ../../libs/gd/gd-2.0.35/jisx0208.h.
 #
