diff -up texlive-20210324-source/texk/web2c/pdftexdir/pdftoepdf.cc.omv~ texlive-20210324-source/texk/web2c/pdftexdir/pdftoepdf.cc
--- texlive-20210324-source/texk/web2c/pdftexdir/pdftoepdf.cc.omv~	2021-09-14 20:04:07.690752768 +0200
+++ texlive-20210324-source/texk/web2c/pdftexdir/pdftoepdf.cc	2021-09-14 20:04:17.476804217 +0200
@@ -967,8 +967,8 @@ The changes below seem to work fine.
 */
 // begin modification
             groupDict = pageDict->lookup("Group");
-            const Dict& dic1 = page->getGroup();
-            const Dict& dic2 = groupDict.getDict();
+            const Dict& dic1 = *page->getGroup();
+            const Dict& dic2 = *groupDict.getDict();
             // replace dic2 in groupDict with dic1
             l = dic2.getLength();
             for (i = 0; i < l; i++) {
