--- texlive-20140525-source/texk/dvidvi/dvidvi.c.p0~	2013-06-28 15:27:05.000000000 +0200
+++ texlive-20140525-source/texk/dvidvi/dvidvi.c	2014-06-16 23:44:00.913963976 +0200
@@ -327,7 +327,7 @@ static void stringdvibuf(integer p, inte
  *   Print a usage error messsage, and quit.
  */
 static void usage(void) {
-   (void)fprintf(stderr,banner);
+   (void)fprintf(stderr,"%s",banner);
    (void)fprintf(stderr,"Usage:  dvidvi [options] input[.dvi] [output]\n");
    (void)fprintf(stderr,"where options are:\n");
    (void)fprintf(stderr,"    [-f n] first page printed     [-l n] last page printed\n");
@@ -806,7 +806,7 @@ default:
       } /* else argument with '-' */
    }  /* for */
    if (*iname == 0) {
-       (void)fprintf(stderr, banner) ;
+       (void)fprintf(stderr, "%s", banner) ;
        error("! no input file specified");
    }
    /* Inserted by djc@dsmail.hmi.de 3.8.1994 */
@@ -817,7 +817,7 @@ default:
    }
 
    if (*oname != 0 && !quiet) {
-       (void)fprintf(stderr, banner) ;
+       (void)fprintf(stderr, "%s", banner) ;
        (void)fprintf(stderr, "%s -> %s\n",iname,oname);
        temp = nextstring ;
    }
--- texlive-20140525-source/texk/web2c/cwebboot.cin.p0~	2014-02-12 09:50:58.000000000 +0100
+++ texlive-20140525-source/texk/web2c/cwebboot.cin	2014-06-16 23:44:00.913963976 +0200
@@ -1092,7 +1092,7 @@ void
  fatal(const char*s,const char*t)
 #line 1182 "cwebdir/common.w"
 {
-if(*s)printf(s);
+if(s&&*s)printf("%s",s);
 err_print(t);
 history= fatal_message;exit(wrap_up());
 }
--- texlive-20140525-source/texk/web2c/cwebdir/common.w.p0~	2012-09-05 15:27:45.000000000 +0200
+++ texlive-20140525-source/texk/web2c/cwebdir/common.w	2014-06-16 23:44:00.913963976 +0200
@@ -1180,7 +1180,7 @@ concatenated to print the final error me
 fatal(s,t)
   char *s,*t;
 {
-  if (*s) printf(s);
+  if (s && *s) printf("%s", s);
   err_print(t);
   history=fatal_message; exit(wrap_up());
 }
--- texlive-20140525-source/texk/web2c/cwebdir/cweave.w.p0~	2012-09-05 15:27:45.000000000 +0200
+++ texlive-20140525-source/texk/web2c/cwebdir/cweave.w	2014-06-16 23:44:00.913963976 +0200
@@ -1784,7 +1784,7 @@ void
 print_cat(c) /* symbolic printout of a category */
 eight_bits c;
 {
-  printf(cat_name[c]);
+  printf("%s", cat_name[c]);
 }
 
 @ The token lists for translated \TEX/ output contain some special control
--- texlive-20140525-source/texk/web2c/omegafonts/out_routines.c.p0~	2009-11-10 11:27:09.000000000 +0100
+++ texlive-20140525-source/texk/web2c/omegafonts/out_routines.c	2014-06-16 23:44:00.914963976 +0200
@@ -383,5 +383,5 @@ out_digits(unsigned counter)
 void
 out(const_string sval)
 {
-	fprintf(file_output, sval);
+	fprintf(file_output, "%s", sval);
 }
--- texlive-20140525-source/texk/web2c/tiedir/tie-w2c.ch.p0~	2014-02-12 09:50:58.000000000 +0100
+++ texlive-20140525-source/texk/web2c/tiedir/tie-w2c.ch	2014-06-16 23:44:00.914963976 +0200
@@ -148,7 +148,7 @@ int main (int argc, string *argv)
 @y
   kpse_set_program_name(argv[0], "tie");
   print(banner); /* print a ``banner line'' */
-  print_ln(versionstring);  /* Web2C version */
+  printf("%s", versionstring);  /* Web2C version */
   print_ln(copyright); /* include the copyright notice */
 @z
 
--- texlive-20150521-source/texk/ps2pk/objects.h.omv~	2016-05-06 21:41:05.533902923 +0200
+++ texlive-20150521-source/texk/ps2pk/objects.h	2016-05-06 21:41:22.680828874 +0200
@@ -231,7 +231,7 @@ struct xobject {
 /*SHARED*/
 /* NDW: personally, I want to see status and error messages! */
 #define IfTrace0(condition,model)                                 \
-        {if (condition) printf(model);}
+        {if (condition) printf("%s",model);}
 #define IfTrace1(condition,model,arg0)                            \
         {if (condition) printf(model,arg0);}
 #define IfTrace2(condition,model,arg0,arg1)                       \
--- texlive-20150521-source/texk/web2c/omegafonts/error_routines.c.omv~	2016-05-06 21:34:20.105561810 +0200
+++ texlive-20150521-source/texk/web2c/omegafonts/error_routines.c	2016-05-06 21:43:44.102223681 +0200
@@ -61,9 +61,7 @@ lex_error_s_1(const_string fmt, const_st
 void
 yyerror(const_string fmt)
 {
-    fprintf(stderr, "line %d (parsing): ", line_number);
-    fprintf(stderr, fmt);
-    fprintf(stderr, "\n");
+    fprintf(stderr, "line %d (parsing): %s\n", line_number, fmt);
     num_errors++;
 }
 
@@ -71,9 +69,7 @@ yyerror(const_string fmt)
 void
 warning_0(const_string fmt)
 {
-    fprintf(stderr, "line %d (warning): ", line_number);
-    fprintf(stderr, fmt);
-    fprintf(stderr, "\n");
+    fprintf(stderr, "line %d (warning): %s\n", line_number, fmt);
 }
 
 void
@@ -119,9 +115,7 @@ warning_s_2(const_string fmt, const_stri
 void
 fatal_error_0(const_string fmt)
 {
-    fprintf(stderr, "line %d (fatal): ", line_number);
-    fprintf(stderr, fmt);
-    fprintf(stderr, "\n");
+    fprintf(stderr, "line %d (fatal): %s\n", line_number, fmt);
     exit(1);
 }
 
@@ -155,9 +149,7 @@ fatal_error_s(const_string fmt, const_st
 void
 internal_error_0(const_string fmt)
 {
-    fprintf(stderr, "line %d (internal): ", line_number);
-    fprintf(stderr, fmt);
-    fprintf(stderr, "\n");
+    fprintf(stderr, "line %d (internal): %s\n", line_number, fmt);
     exit(2);
 }
 
--- texlive-20150521-source/utils/devnag/src/devnag.c.omv~	2016-05-06 21:00:44.817976048 +0200
+++ texlive-20150521-source/utils/devnag/src/devnag.c	2016-05-06 21:00:52.054946597 +0200
@@ -2400,7 +2400,7 @@ void put_macro(short macro) {
 void err_ill(const char *str) {
    fprintf(stderr, "Error: illegal character(s) \"%s\" detected at line %d:\n",
 	  str, linenumber);
-   fprintf(stderr, inbuf);
+   fprintf(stderr, "%s", inbuf);
    exit(1);
 }
 
