diff -up texlive-20260301-source/texk/dvisvgm/dvisvgm-src/libs/clipper/clipper.cpp.5~ texlive-20260301-source/texk/dvisvgm/dvisvgm-src/libs/clipper/clipper.cpp
--- texlive-20260301-source/texk/dvisvgm/dvisvgm-src/libs/clipper/clipper.cpp.5~	2018-09-13 19:10:25.000000000 +0200
+++ texlive-20260301-source/texk/dvisvgm/dvisvgm-src/libs/clipper/clipper.cpp	2026-03-03 13:53:34.966544996 +0100
@@ -1970,7 +1970,7 @@ void Clipper::DeleteFromSEL(TEdge *e)
 #ifdef use_xyz
 void Clipper::SetZ(IntPoint& pt, TEdge& e1, TEdge& e2)
 {
-  if (pt.Z != 0 || !m_ZFill) return;
+  if (pt.Z || !m_ZFill) return;
   else if (pt == e1.Bot) pt.Z = e1.Bot.Z;
   else if (pt == e1.Top) pt.Z = e1.Top.Z;
   else if (pt == e2.Bot) pt.Z = e2.Bot.Z;
diff -up texlive-20260301-source/texk/dvisvgm/dvisvgm-src/src/Unicode.cpp.5~ texlive-20260301-source/texk/dvisvgm/dvisvgm-src/src/Unicode.cpp
--- texlive-20260301-source/texk/dvisvgm/dvisvgm-src/src/Unicode.cpp.5~	2026-01-20 00:07:15.000000000 +0100
+++ texlive-20260301-source/texk/dvisvgm/dvisvgm-src/src/Unicode.cpp	2026-03-03 13:54:43.242191069 +0100
@@ -185,7 +185,7 @@ uint32_t Unicode::toSurrogate (uint32_t
 
 uint32_t Unicode::toLigature (const string &nonlig) {
 	struct Ligature {
-		const char *nonlig;
+		const char8_t *nonlig;
 		uint32_t lig;
 	} ligatures[39] = {
 		{u8"AA",  0xA732}, {u8"aa", 0xA733},
@@ -215,7 +215,7 @@ uint32_t Unicode::toLigature (const stri
 		{u8"VY",  0xA760}, {u8"tz",  0xA761},
 	};
 	auto it = algo::find_if(ligatures, [&nonlig](const Ligature &l) {
-		return l.nonlig == nonlig;
+		return reinterpret_cast<const char *>(l.nonlig) == nonlig;
 	});
 	return it != end(ligatures) ? it->lig : 0;
 }
diff -up texlive-20260301-source/texk/dvisvgm/dvisvgm-src/src/fonts/Base14Fonts.hpp.5~ texlive-20260301-source/texk/dvisvgm/dvisvgm-src/src/fonts/Base14Fonts.hpp
--- texlive-20260301-source/texk/dvisvgm/dvisvgm-src/src/fonts/Base14Fonts.hpp.5~	2025-01-12 15:58:51.000000000 +0100
+++ texlive-20260301-source/texk/dvisvgm/dvisvgm-src/src/fonts/Base14Fonts.hpp	2026-03-03 13:53:34.966964651 +0100
@@ -25,6 +25,7 @@
 
 struct MemoryFontData {
 	MemoryFontData () =delete;
+	MemoryFontData(const char *d, size_t s):data(d),size(s) {}
 	const char *data;
 	size_t size;
 };
