--- texlive-20131212-source/utils/asymptote/settings.cc.orig	2013-12-12 03:50:50.546435324 -0200
+++ texlive-20131212-source/utils/asymptote/settings.cc	2013-12-12 03:51:08.375435650 -0200
@@ -100,7 +100,7 @@ string defaultPSViewer="gv";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
 string defaultDisplay="display";
